README.md

```diff
@@ -145,21 +148,25 @@ does not have a `[[Call]]` internal method; it is not possible to invoke the
 When called with arguments `value1` and `value2`:
 
 * If `Type(value1)` is not `Object` return `false`.
-* Let `B` be `? value1.[[GetOwnProperty]](@@builtin)`.
-* If `B` is `undefined`, return `false`.
-* Let `V1` be `? GET(value1, @@builtin)`.
-* If `Type(V1)` is not `String`, return `false`.
+* If `? [[HasOwnProperty]](value1, @@builtin)` is `false`, return `false`.
+* Let `F1` be `? GetMethod(value1, @@builtin)`.
+* If `F1` is `undefined`, return `false`.
+* Let `V1` be `? Call(F1, value1)`.
+* If `V1` is `undefined`, return `false`.
 * If `value2` is `undefined`, return `true`.
 * If `Type(value2)` is not `Object`, return `false`.
-* Let `B` be `? value2.[[GetOwnProperty]](@@builtin)`.
-* If `B` is `undefined`, return `false`.
-* Let `V2` be `? GET(value2, @@builtin)`.
-* If `Type(V2)` is not `String`, return `false`.
-* Let `same` be the result of performing Strict Equality Comparison `V1 === V2`.
+* If `? [[HasOwnProperty]](value2, @@builtin)` is `false`, return `false`.
+* Let `F2` be `? GetMethod(value2, @@builtin)`.
+* If `F2` is `undefined`, return `false`.
+* Let `V2` be `? Call(F2, value2)`.
+* If `V2` is `undefined`, return `false`.
+* Let `same` be the result of performing Strict Equality Comparison
```

Comment:
The ToString could throw, so I think that should be in its own algorithm steps with ?